### PR TITLE
fix(ui5-carousel): hides navigation arrows on mouse out

### LINKED ISSUE #3136: Carousel: Behavior improvement
**Describe the bug**
Some behavior are not the same as UI5 Carousel control

**Isolated Example**
https://sap.github.io/ui5-webcomponents-react/?path=/docs/layouts-floorplans-carousel--default-story

**To Reproduce**
1. Go to https://sap.github.io/ui5-webcomponents-react/?path=/docs/layouts-floorplans-carousel--default-story

**Expected behavior**
Expect the same behavior as UI5
1. When mouse move out of the carousel area, the navigation button should be hidden.
2. Padding of navigation button suppose to be the same as UI5 Carousel control
3. Need property support to hide the pagination.
https://sapui5.hana.ondemand.com/#/entity/sap.m.Carousel/sample/sap.m.sample.CarouselWithDisplayOptions

**Screenshots**
image

**UI5 Web Components for React Information**
`@ui5/webcomponents` version: Currently Version
`@ui5/webcomponents-react` version: Currently Version
Operating System: Mac OS
Browser: Chrome

**Additional context**
I am from SuccessFactors, and current we are using Carousel to build project feature while the behavior are not the same as UI5 control.

### PR DESCRIPTION
**Pull request by kskondov**
Fixes #3136

### CODE REVIEW

File: packages/main/src/Carousel.js [hunk 1]
```diff
@@ -152,6 +152,14 @@ const metadata = {
 		_itemWidth: {
 			type: Integer,
 		},
+
+		/**
+		 * If set to true navigation arrows are hidden
+		 * @private
+		 */
+		_hideNavigationArrows: {
```
Comment (gmkv) [inline packages/main/src/Carousel.js:160]: What is the difference between this property and the existing property hideNavigation?

**Review by gmkv — CHANGES_REQUESTED** [review]
On initial render, the arrows are still visible.
They should show up only when the user's mouse goes over the carousel.

File: packages/main/src/Carousel.js (hunk 1 above)
Comment (kskondov) [reply packages/main/src/Carousel.js:160]: hideNavigation hides the whole navigation including the pagination

Comment (gmkv) [reply packages/main/src/Carousel.js:160]: Ok. Use `noAttribute: true` on this property which will prevent the rendering of a `_hide-navigation-arrows` attribute to the element. You don't need to pollute the DOM with such internal stuff.

File: packages/main/src/Carousel.js [hunk 2]
```diff
@@ -472,7 +489,7 @@ class Carousel extends UI5Element {
 	}
 
 	get arrows() {
-		const showArrows = this.showNavigationArrows && isDesktop();
+		const showArrows = this._visibleNavigationArrows && this.showNavigationArrows && isDesktop();
```
Comment (gmkv) [inline packages/main/src/Carousel.js:492]: Now `_visibleNavigationArrows` and `showNavigationArrows` sound too similar.

I see that `showNavigationArrows` does more than just show the arrows - it also controls displaying the dots.
It makes more sense to rename it to `showNavigation`.

**Review by gmkv — CHANGES_REQUESTED** [review]

File: packages/main/src/Carousel.js [hunk 3]
```diff
@@ -152,6 +152,15 @@ const metadata = {
 		_itemWidth: {
 			type: Integer,
 		},
+
+		/**
+		 * If set to true navigation arrows are shown
+		 * @private
```
Comment (fifoosid) [inline packages/main/src/Carousel.js:158]: add since tag

**Review by fifoosid — CHANGES_REQUESTED** [review]

**Review by gmkv — APPROVED** [review]

**Pull request merged**